v5.11.1 (#4935)
mui/mui-x: CHANGELOG.md

All notable changes to this project will be documented in this file.
See [Conventional Commits](https://conventionalcommits.org) for commit guidelines.

## v5.11.1

> _May 29, 2022_

Reviewer: ```suggestion
_May 20, 2022_
```

_May 29, 2022_

We'd like to offer a big thanks to the 6 contributors who made this release possible. Here are some highlights ✨:

> - 🌍 Support localization on the date and time picker components (#4517) @alexfauquette

Reviewer 2: Putting a working code will take a lot of space, maybe a link to the doc is enough. If in the next release we bring full support of localization, it will be time to ask for help in translation aria texts

```suggestion
- 🌍 Support localization on the date and time picker components (#4517) @alexfauquette

	You can translate some texts as you do with the data grid. Check the [localization documentation for more information](mui.com/x/react-date-pickers/localization/)
```

> - 🌍 Support localization on the date and time picker components (#4517) @alexfauquette

Author: @alexfauquette if you want to add a small code example

### `@mui/x-data-grid@v5.11.1` / `@mui/x-data-grid-pro@v5.11.1` / `@mui/x-data-grid-premium@v5.11.1`

#### Changes

- [DataGrid] Add class corresponding to current density (#4858) @m4theushw
- [DataGrid] Execute the pipe-processors in their initialization order (#4913) @flaviendelangle
- [DataGrid] Fix rendering of the no rows overlay when the `loading` prop is changed (#4910) @m4theushw

> - [DataGridPremium] Support column spanning in Excel export (#4830) @cherniavskii

Reviewer 2: To highlight with the calendar screenshot provided in the PR?

- [l10n] Improve Russian (ru-RU) locale (#4864) @arvkonstantin

### `@mui/x-date-pickers@5.0.0-alpha.4` / `@mui/x-date-pickers-pro@5.0.0-alpha.4`

#### Changes

- [DatePicker] Fix keyboard accessibility for first and last year (#4807) @alexfauquette
- [pickers] Add l10n support (#4517) @alexfauquette
- [pickers] Close Popper when pressing <kbd>Esc</kbd> inside a modal (#4499) @alexfauquette
- [pickers] Support class slots on toolbar components (#4855) @flaviendelangle

> - [TimePicker] Fix time validation when current date is null (#4867) @flaviendelangle

Reviewer 2: ```suggestion
- [TimePicker] Fix time validation when current date is `null` (#4867) @flaviendelangle
```

### Docs

- [docs] Avoid confusion with license key installation (#4891) @oliviertassinari
- [docs] Complete the instructions for pickers installation in readme (#4852) @alexfauquette
- [docs] Disable ads on paid-only pages (#4842) @flaviendelangle

> - [docs] Don't redirect on deploy preview (#4818) @m4theushw

Reviewer: ```suggestion
- [docs] Don't redirect to localized doc on deploy preview (#4818) @m4theushw
```

- [docs] Limit `LICENSE` file to 80 char per line (#4873) @oliviertassinari
- [docs] Typo on OrderId @oliviertassinari
- [docs] Update feature comparison table (#4918) @cherniavskii

### Core

- [core] Add new script to generate tree data rows from file tree (#4902) @flaviendelangle

> - [core] Code style edit (#4874) @oliviertassinari

Reviewer: ```suggestion
- [core] Fix code style (#4874) @oliviertassinari
```

All notable changes to this project will be documented in this file.
See [Conventional Commits](https://conventionalcommits.org) for commit guidelines.

## v5.11.1

_May 20, 2022_

We'd like to offer a big thanks to the 6 contributors who made this release possible. Here are some highlights ✨:

- 🌍 Support localization on the date and time picker components (#4517) @alexfauquette

  You can translate some texts as you do with the data grid. Check the [localization documentation for more information](mui.com/x/react-date-pickers/localization/)

- 📃 Support column spanning in the Excel export (#4830) @cherniavskii

  <img src="https://user-images.githubusercontent.com/13808724/167691417-bf6baeb9-d409-4134-acb6-aadaf6434de9.png" width="800">

- 🐞 Bugs fixes

### `@mui/x-data-grid@v5.11.1` / `@mui/x-data-grid-pro@v5.11.1` / `@mui/x-data-grid-premium@v5.11.1`

#### Changes

- [DataGrid] Add class corresponding to current density (#4858) @m4theushw
- [DataGrid] Execute the pipe-processors in their initialization order (#4913) @flaviendelangle
- [DataGrid] Fix rendering of the no rows overlay when the `loading` prop is changed (#4910) @m4theushw

> - [DataGridPremium] Support column spanning in the Excel export (#4830) @cherniavskii

Reviewer 2: ```suggestion
- [DataGridPremium] Add exceljs to the dependencies (#4939) @alexfauquette
- [DataGridPremium] Support column spanning in the Excel export (#4830) @cherniavskii
```

- 🐞 Bugs fixes

### `@mui/x-data-grid@v5.11.1` / `@mui/x-data-grid-pro@v5.11.1` / `@mui/x-data-grid-premium@v5.11.1`

#### Changes

- [DataGrid] Add class corresponding to current density (#4858) @m4theushw
- [DataGrid] Execute the pipe-processors in their initialization order (#4913) @flaviendelangle
- [DataGrid] Fix rendering of the no rows overlay when the `loading` prop is changed (#4910) @m4theushw
- [DataGridPremium] Add `exceljs` to the dependencies (#4939) @alexfauquette
- [DataGridPremium] Support column spanning in the Excel export (#4830) @cherniavskii
- [l10n] Improve Russian (ru-RU) locale (#4864) @arvkonstantin

### `@mui/x-date-pickers@5.0.0-alpha.4` / `@mui/x-date-pickers-pro@5.0.0-alpha.4`

#### Changes

- [DatePicker] Fix keyboard accessibility for first and last year (#4807) @alexfauquette
- [pickers] Add l10n support (#4517) @alexfauquette
- [pickers] Close Popper when pressing <kbd>Esc</kbd> inside a modal (#4499) @alexfauquette
- [pickers] Support class slots on toolbar components (#4855) @flaviendelangle
- [TimePicker] Fix time validation when current date is `null` (#4867) @flaviendelangle

### Docs

> - [docs] Avoid confusion with license key installation (#4891) @oliviertassinari

Reviewer 2: ```suggestion
- [docs] Add 301 redirect for columns page (#4940) @alexfauquette
- [docs] Avoid confusion with license key installation (#4891) @oliviertassinari
```

All notable changes to this project will be documented in this file.
See [Conventional Commits](https://conventionalcommits.org) for commit guidelines.

## v5.11.1

_May 20, 2022_

We'd like to offer a big thanks to the 6 contributors who made this release possible. Here are some highlights ✨:

- 🌍 Support localization on the date and time picker components (#4517) @alexfauquette

>   You can translate some texts as you do with the data grid. Check the [localization documentation for more information](mui.com/x/react-date-pickers/localization/)

Reviewer 3: ```suggestion
  Texts can be translated in the pickers components, similar to what can be done in the data grid component. Check the [localization documentation for more information](mui.com/x/react-date-pickers/localization/)
```

>   Texts can be translated in the pickers components, similar to what can be done in the data grid component. Check the [localization documentation for more information](mui.com/x/react-date-pickers/localization/)

Reviewer: This link is too big.

```suggestion
  Texts can be translated in the pickers components, similar to what can be done in the data grid component. Check the [documentation](https://mui.com/x/react-date-pickers/localization/) for more information.
```

> - 📃 Support column spanning in the Excel export (#4830) @cherniavskii

Reviewer 3: ```suggestion
- 📃 Add support for column spanning when excel exporting (#4830) @cherniavskii
```

Author: ```suggestion
- 📃 Add support for column spanning when exporting for Excel (#4830) @cherniavskii
```

"when Excel exporting" is weird

Reviewer 3: This also sounds good.

> - 📃 Add support for column spanning when exporting for Excel (#4830) @cherniavskii

Reviewer: ```suggestion
- 📃 Add support for column spanning when exporting to Excel (#4830) @cherniavskii
```

  <img src="https://user-images.githubusercontent.com/13808724/167691417-bf6baeb9-d409-4134-acb6-aadaf6434de9.png" width="800">

- 🐞 Bugs fixes

### `@mui/x-data-grid@v5.11.1` / `@mui/x-data-grid-pro@v5.11.1` / `@mui/x-data-grid-premium@v5.11.1`

#### Changes

> - [DataGrid] Add class corresponding to current density (#4858) @m4theushw

Reviewer 3: ```suggestion
- [DataGrid] Add a class corresponding to current density (#4858) @m4theushw
```

  Texts can be translated in the pickers components, similar to what can be done in the data grid component. Check the [documentation](https://mui.com/x/react-date-pickers/localization/) for more information.

- 📃 Add support for column spanning when exporting to Excel (#4830) @cherniavskii

  <img src="https://user-images.githubusercontent.com/13808724/167691417-bf6baeb9-d409-4134-acb6-aadaf6434de9.png" width="800">

- 🐞 Bugs fixes

### `@mui/x-data-grid@v5.11.1` / `@mui/x-data-grid-pro@v5.11.1` / `@mui/x-data-grid-premium@v5.11.1`

#### Changes

- [DataGrid] Add a CSS class corresponding to current density (#4858) @m4theushw
- [DataGrid] Execute the pipe-processors in their initialization order (#4913) @flaviendelangle
- [DataGrid] Fix rendering of the no rows overlay when the `loading` prop is changed (#4910) @m4theushw
- [DataGridPremium] Add `exceljs` to the dependencies (#4939) @alexfauquette
- [DataGridPremium] Support column spanning in the Excel export (#4830) @cherniavskii
- [l10n] Improve Russian (ru-RU) locale (#4864) @arvkonstantin

### `@mui/x-date-pickers@5.0.0-alpha.4` / `@mui/x-date-pickers-pro@5.0.0-alpha.4`

#### Breaking changes

> The following props have been removed: `clearable`, `showTodayButton`, `cancelText`, `okText`.

Author: - The props related to the action bar buttons have been removed (`clearable`, `showTodayButton`, `cancelText`, `okText`)

    To decide which button must be displayed and in which order, you can now use the `actions` prop of the `actionBar` component slot props.

    ```jsx
    <DatePicker
      // The actions will be the same between desktop and mobile
      componentsProps={{ actionBar: { actions: ['clear'] } }}

      // The actions will be the different between desktop and mobile
      componentsProps={{ actionBar: { actions: (variant) => variant === 'desktop' ? [] : ['clear'] } }}
    />
    ```

    The build-it `ActionBar` component supports 4 different actions: `'clear'`, `'cancel'`, `'accept'`, and `'today'`.
    The default values are `['cancel', 'accept']` on mobile and `[]` on desktop.

    If you need other actions, you can provide your own component to the `ActionBar` component slot

    ```jsx
    <DatePicker
      componentsProps={{ ActionBar: CustomActionBar }}
    />
    ```

Reviewer 2: Yes, showing the two ways to use the `actions` prop is a better idea than explaining it.

I don't see ways to improve the structure. I only found two typos

```diff
-// The actions will be the different between desktop and mobile
+// The actions will be different between desktop and mobile
```

```diff
-The build-it `ActionBar`
+The built-in `ActionBar`
```

Author: Applied

To control the presence of action buttons, you can pass props to the `ActionBar` as follow

```jsx
<DatePicker
  componentsProps={{ actionBar: { actions: ['clear'] } }}
/>
```

The default `ActionBar` accept a prop `actions` which indicates buttons to display and their order.
Default values are `['cancel', 'accept']` for mobile and `[]` for desktop.

If you want to have a different set of actions for mobile and desktop components, `actions` can be a callback.
It takes as an input the component variant and returns the array of actions to display.

```jsx
<DatePicker
  // This example show no action on desktop, and 'clear' action otherwise
  componentsProps={{ actionBar: { actions: (variant) => variant === 'desktop' ? [] : ['clear'] } }}
/>
```

Available actions are `'clear'`, `'cancel'`, `'accept'`, and `'today'`.

> You can add other ones by overriding the `ActionBar` slot.

Reviewer 2: Added the breaking change in a PR, because ` ``` ` were interfeting
